File: python/8_kyu/holiday_vi_shark_pontoon.py
def shark(pontoon_distance, shark_distance, you_speed, shark_speed, dolphin):
    if dolphin:
        shark_time = shark_distance / (shark_speed / 2)
    else:
        shark_time = shark_distance/shark_speed
        
    my_time = pontoon_distance/you_speed
    
    if my_time < shark_time:
        return "Alive!"
    else:
        return "Shark Bait!"
    


shark(12, 50, 4, 8, True)

File: javascript/8_kyu/holiday_vi_shark_pontoon.js
// Holiday VI - Shark Pontoon
// https://www.codewars.com/kata/57e921d8b36340f1fd000059/train/javascript

function shark(pontoonDistance, sharkDistance, youSpeed, sharkSpeed, dolphin){
    // time = distance / rate
      var to_live = 0;
      var give_up = 0;
      
      if (dolphin) {
        give_up += sharkDistance / (sharkSpeed / 2);
      } else {
        give_up += sharkDistance / sharkSpeed;
      }
      
      to_live += pontoonDistance / youSpeed;
      
      if (to_live < give_up) {
        return "Alive!";
      } else {
        return "Shark Bait!";
      }
    }